core/controller/src/main/scala/org/apache/openwhisk/core/controller/Actions.scala

```diff
@@ -218,9 +229,36 @@ trait WhiskActionsApi extends WhiskCollectionAPI with PostActionActivation with
 
         onComplete(checkAdditionalPrivileges) {
           case Success(_) =>
-            putEntity(WhiskAction, entityStore, entityName.toDocId, overwrite, update(user, request) _, () => {
-              make(user, entityName, request)
-            })
+            onComplete(WhiskActionVersionList.get(entityName, entityStore)) {
+              case Success(result) =>
+                val id = result.matchedDocId(content.version).getOrElse(entityName.toDocId)
+                putEntity(
+                  WhiskAction,
+                  entityStore,
+                  id,
+                  true,
+                  update(user, request) _,
+                  () => {
+                    make(user, entityName, request)
+                  },
+                  postProcess = Some { action: WhiskAction =>
+                    // delete oldest version when created successfully
```

Comment:
Can we have an operator level configuration where if you are at the max limit of versions, you either do this delete the oldest version or reject the create request and require the user to delete an old version before creating a new one?

Reply:
good idea, I will update

Reply:
add a `deleteOld` parameter to `create` method